From 309b19d942c05572fd2dd03b592b7323e063c0e6 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Sun, 24 Dec 2023 19:59:23 +0800
Subject: [PATCH] apple_utils.c: should be included for Darwin, not for Cocoa

The build needs this file for linking even when Cocoa is disabled.
Commit which introduced it mistakenly includes it only for Cocoa:
https://github.com/mpv-player/mpv/commit/0463096b3c5a717a7c181801c8df797b046989fc

As of 0.40.0, linking works without it, but related runtime error happens in some cases.

diff --git meson.build meson.build
index f9fe4e7263..a8b91061a0 100644
--- meson.build
+++ meson.build
@@ -409,7 +409,6 @@
     dependencies += cocoa
     sources += files('osdep/language-mac.c',
                      'osdep/path-mac.m',
-                     'osdep/utils-mac.c',
                      'osdep/mac/app_bridge.m',
                      'player/clipboard/clipboard-mac.m')
     main_fn_source = files('osdep/main-fn-mac.c')
@@ -434,6 +433,7 @@
 endif
 
 if darwin
+    sources += files('osdep/utils-mac.c')
     path_source = files('osdep/path-darwin.c')
     timer_source = files('osdep/timer-darwin.c')
 endif
